### ISSUE TITLE
Custom image

### ISSUE THREAD

**Issue by hbinol** [HavenFeng/photometric_optimization#7]
Nice work!

I wonder how can I apply this for a custom image?

I guess I need both 

- a landmark file ( landmarkpath = os.path.sep.join([input_folder, image_name + '.npy']) )
- image mask file ( image_mask_path = os.path.sep.join([image_mask_folder, image_name + '.npy']) )

for the custom image, right? Do I need more than this? How can I obtain these files for a custom image?

Thanks!

**Comment by HavenFeng** [comment]
Yes, you would need both, as we've described in the readme file.

**Comment by hbinol** [comment]
Thank you for your answer.

The next question is how can I generate these .npy files? Is there any sample script for those applications (face_alignment & face_seg)?

**Comment by JacksonL1** [comment]
> Thank you for your answer.
> 
> The next question is how can I generate these .npy files? Is there any sample script for those applications (face_alignment & face_seg)?

[my fork](https://github.com/JacksonL1/photometric_optimization) solves the custom image

**Comment by hbinol** [comment]
Thank you so much @JacksonL1

Do you have any idea about how I can pick a point (or points) on the neck of the initial 3D mesh template and follow its trajectory during photometric optimization? In a nutshell, I want to know where are the neck points of the final mesh compared to 2D or 3D landmarks.

**Comment by JacksonL1** [comment]
> Thank you so much @JacksonL1
> 
> Do you have any idea about how I can pick a point (or points) on the neck of the initial 3D mesh template and follow its trajectory during photometric optimization? In a nutshell, I want to know where are the neck points of the final mesh compared to 2D or 3D landmarks.

I just know that at [face-parsing.Pytorch](https://github.com/zllrunning/face-parsing.PyTorch) project, the 14th part is neck. 
use this part to generate the mask and calculate the photometric loss. 
Hope answered your question

**Comment by hbinol** [comment]
That's great! Thank you @JacksonL1 

I need to solve the problem of distance calculation between a neck point and chin on the final 3D mesh.

**Comment by JacksonL1** [comment]
I don't know how to get points on the neck. I'm sorry about that~

**Comment by hbinol** [comment]
Hi @JacksonL1 

For your work, I got an error like below

 No such file or directory: './model/s3fd.pth'

I found that later, but now there is another missing file

/model/face_seg.pth

**Comment by JacksonL1** [comment]
hi @hbinol 
you can find the model at [link](https://pan.baidu.com/s/1S-CYb3KFk2CI02HU_3jUKA) extract code：1emq

**Comment by JacksonL1** [comment]
@hbinol 
if you can not use the Baidu Cloud, also can find the model at [Google Cloud](https://drive.google.com/file/d/1_vyhJUiy3y9DyrtPRkqBq00nkpn766c-/view?usp=sharing)

**Comment by hbinol** [comment]
I need access to Google Cloud. I believe I've already requested access.

**Comment by hbinol** [comment]
@JacksonL1 I need access to Google Cloud. I believe I've already requested access.

**Comment by JacksonL1** [comment]
@hbinol download [official pre-train model](https://drive.google.com/file/d/154JgKpzCPW82qINcVieuPH3fZ2e0P812/view) and rename to face_seg.pth

**Comment by yukangcao** [comment]
> @hbinol download [official pre-train model](https://drive.google.com/file/d/154JgKpzCPW82qINcVieuPH3fZ2e0P812/view) and rename to face_seg.pth

Hi @JacksonL1 Thanks for your work.
I just met a problem 'Expected object  of scalar double but got scalar type float for sequence element 4 when running your package. It comes from 'wj_fitting.py, line 175, in optimize'. Do you have any idea about this? Thank you.

**Comment by JacksonL1** [comment]
> > @hbinol download [official pre-train model](https://drive.google.com/file/d/154JgKpzCPW82qINcVieuPH3fZ2e0P812/view) and rename to face_seg.pth
> 
> Hi @JacksonL1 Thanks for your work.
> I just met a problem 'Expected object of scalar double but got scalar type float for sequence element 4 when running your package. It comes from 'wj_fitting.py, line 175, in optimize'. Do you have any idea about this? Thank you.

I just run the code, it will run smoothly. 
The error seems like data type is wrong. Could you show me the code near by 175 and full error message?

**Comment by yukangcao** [comment]
Thanks for your reply. Please check the attachment for the original error
message.
By the way, I just changed all the values of grids (wj_fitting.py, line
175) to be double type, and the problem was solved this way.
Thank you and have a nice day!

Best regards,
Yukang

**Comment by pfeducode** [comment]
> > 谢谢您的回答。
> > 下一个问题是如何生成这些 .npy 文件？这些应用程序（face_alignment 和 face_seg）是否有任何示例脚本？
> 
> [我的叉子](https://github.com/JacksonL1/photometric_optimization)解决了自定义图像

How to change the color of 3D mesh
